src/ActionMailerNext.Standalone/Helpers/HandlebarsHelpers.cs

```diff
@@ -0,0 +1,1206 @@
+                {
+                    routeValues.Add("action", actionName);
+                    routeValues.Add("controller", controllerName);
+
+                    uri = UtilHelper.BuildURI(protocol ?? _viewSettings.Protocol, hostName ?? _viewSettings.Hostname, _viewSettings.UrlPattern, routeValues);
+                }
+                else
+                {
+                    var uriBuilder = new UriBuilder(protocol ?? _viewSettings.Protocol, hostName ?? _viewSettings.Hostname)
+                    {
+                        Path = controllerName + "/" + actionName
+                    };
+                    uri = uriBuilder.Uri;
+                }
+
+                var link = uri.ToString();
+
+                return link;
+            });
+        }
+
+        /// <summary>
+        /// {{stringFormat '{0} {1] {2]' 'Hello' 'World' '!'}}
+        /// </summary>
+        /// <return>
+        /// Formatted string
+        /// </return>
+        public void RegisterStringFormat_Helper()
+        {
+            _hbsService.RegisterHelper("stringFormat", (output, context, arguments) =>
+            {
+                if (arguments.Length < 1)
+                {
+                    throw new HandlebarsException("{{stringFormat}} helper must have at least 1 argument");
+                }
+
+                var args = new string[arguments.Length - 1];
+                for (var i = 0; i < args.Length; i++)
+                {
+                    args[i] = arguments[i + 1]?.ToString() ?? string.Empty;
+                }
+
+                output.WriteSafeString(string.Format(arguments[0].ToString(), args));
+            });
+        }
+
+        /// <summary>
+        /// {{resource 'ResourceFileName.Key'}}
+        /// by default it searches for resource file in Xv.Infrastructure.Standard.Resources.(resourceFileName).(key) otherwise it should be used like
+        /// {{resource 'Namespace.ResourceFileName.Key'}}
+        /// 
+        /// It searches for resources in 'de-de' culture file, if it didn't find it then it searches without specifying the culture(default resource file)
+        /// </summary>
+        /// <return>
+        /// Resource's value as string
+        /// </return>
+        public void RegisterResourceResolver_Helper()
+        {
+            _hbsService.RegisterHelper("resource", (output, context, arguments) =>
+            {
+                if (arguments.Length != 1)
+                {
+                    throw new HandlebarsException("{{resource}} helper must have 1 argument");
+                }
+
+                var resource = arguments[0].ToString().Split('.');
+                var resourceFile = String.Join(".", resource.Take(resource.Length - 1));
+                var resourceKey = resource.Last();
+
+                if (resource.Length == 2)
+                {
+                    resourceFile = "Xv.Infrastructure.Standard.Resources." + resourceFile;
```

Comment:
Maybe we could pass this namespace as an argument instead of hardcode it here. In the future we will drop this "Standard" from the namespace and will have to return here

Reply:
You can, it is already mentioned in a comment above this method